xcm-support/src/lib.rs

```diff
@@ -89,22 +89,22 @@ impl<Network: Get<NetworkId>, AccountId: From<[u8; 32]> + Into<[u8; 32]> + Clone
 		let id = match location {
 			MultiLocation {
 				parents: 1,
-				interior: X1(AccountId32 {
-					id,
-					network: NetworkId::Any,
-				}),
-			} => id,
+				interior: X1(AccountId32 { id, network }),
+			} if network == Network::get() => id,
```

Comment:
You needed to _add_ a match arm for the case where `network` is equal to `Network::get()`, not _modify_ the existing one where `network` is set to `Any`. Right now it's missing exactly that case, see https://github.com/paritytech/polkadot/blob/903c6f2f014bcedb9d6be0cae86e6a4a922f692b/xcm/xcm-builder/src/location_conversion.rs#L109-L116 for more information.